src/details/ArborX_Dendrogram.hpp

```diff
@@ -33,12 +34,47 @@ struct Dendrogram
   {
     Kokkos::Profiling::pushRegion("ArborX::Dendrogram::Dendrogram");
 
-    using ConstEdges = Kokkos::View<Details::WeightedEdge const *, MemorySpace>;
-    Details::dendrogramUnionFind(exec_space, ConstEdges(edges), _parents,
-                                 _parent_heights);
+    auto const num_edges = edges.size();
+    auto const num_vertices = num_edges + 1;
+
+    KokkosExt::reallocWithoutInitializing(exec_space, _parents,
+                                          num_edges + num_vertices);
+    KokkosExt::reallocWithoutInitializing(exec_space, _parent_heights,
+                                          num_edges);
+
+    Kokkos::View<Details::UnweightedEdge *, MemorySpace> unweighted_edges(
+        Kokkos::view_alloc(exec_space, Kokkos::WithoutInitializing,
+                           "ArborX::Dendrogram::unweighted_edges"),
+        num_edges);
+    splitEdges(exec_space, edges, unweighted_edges, _parent_heights);
```

Comment:
Do you actually plan on using that function elsewhere?

Reply:
No, but i cant launch parallel_for from within thr constructor.

Reply:
Oh right.